desc/objectives/_stability.py

```diff
@@ -367,3 +369,246 @@ def compute(self, params, constants=None):
             profiles=constants["profiles"],
         )
         return constants["transforms"]["grid"].compress(data["magnetic well"])
+
+
+class BallooningStability(_Objective):
+    """A type of ideal MHD instability.
+
+    Infinite-n ideal MHD ballooning modes are of significant interest.
+    These instabilities are also related to smaller-scale kinetic instabilities.
+    With this class, we optimize MHD equilibria against the ideal ballooning mode.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium that will be optimized to satisfy the Objective.
+    target : {float, ndarray}, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        Must be broadcastable to Objective.dim_f.
+    bounds : tuple of {float, ndarray, callable}, optional
+        Lower and upper bounds on the objective. Overrides target.
+        Both bounds must be broadcastable to to Objective.dim_f
+        If a callable, each should take a single argument `rho` and return the
+        desired bound (lower or upper) of the profile at those locations.
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+        Not used since the growth rate is always normalized.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _coordinates = "r"
+    _units = "(normalized)"
+    _print_value_fmt = "Ideal-ballooning Stability: {:10.3e} "
```

Comment:
This might be clearer as `Ideal-ballooning growth rate: {:10.3e}`, since then  it will print like 
```
Maximum Ideal-ballooning growth rate: 1.2
Average Ideal-ballooning growth rate: 0.1
Minimum Ideal-ballooning growth rate: -1.3
```

for cases in which the user wants to penalize all of them, not just the max

Reply:
It's not the growth rate.

Reply:
i thought it was? What is it exactly then?

Reply:
Just an objective, depends on the max eigenvalue which is the square of the growth rate.